stamina-testkit/src/main/scala/stamina/testkit/StaminaTestKit.scala

```diff
@@ -0,0 +1,85 @@
+package stamina
+package testkit
+
+import scala.util._
+import scala.annotation.tailrec
+
+trait StaminaTestKit { self: org.scalatest.WordSpecLike ⇒
+
+  val defaultSampleId = "default"
+  case class PersistableSample(sampleId: String, persistable: AnyRef, description: Option[String]) {
+    override def toString = persistable.getClass.getSimpleName + description.map(" " + _).getOrElse("")
+  }
+
+  def sample(persistable: AnyRef) = new PersistableSample(defaultSampleId, persistable, None)
+  def sample(sampleId: String, persistable: AnyRef) = new PersistableSample(sampleId, persistable, Some(sampleId))
+  def sample(sampleId: String, persistable: AnyRef, description: String) = new PersistableSample(sampleId, persistable, Some(description))
+
+  implicit class TestablePersisters(persisters: Persisters) extends org.scalatest.Matchers {
+    def generateTestsFor(samples: PersistableSample*): Unit = {
+      samples.foreach { sample ⇒
+        performRoundtrip(sample)
+        deserializeStoredVersions(sample)
+      }
+      // Here we could verify test coverage
+    }
+
+    private def performRoundtrip(sample: PersistableSample) = {
+      s"persist and unpersist $sample" in {
+        persisters.canPersist(sample.persistable) should be(true)
```

Comment:
The unfortunate problem with such an assertion is that the failure case will show up as `false was not equal to true`, which is not very descriptive. There are several ways around this but I guess the best solution would be a simple custom Matcher that would produce a descriptive error message. Perhaps something for later?